### PR TITLE
Update to gradle 8.11

### PR DESCRIPTION
**Pull request by ThadHouse**
Includes much nicer warning APIs. Includes a lot of deperecations, so we need to test things

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -52,6 +52,10 @@ allprojects {
     tasks.withType(Javadoc) {
         options.addBooleanOption('Xdoclint:all,-missing', true)
     }
+
+    tasks.withType(JavaCompile) {
+      options.compilerArgs << "-Xlint:unchecked" << "-Xlint:deprecation"
```
Comment (sciencewhiz) [inline build.gradle:57]: temporary?

Comment (ThadHouse) [reply build.gradle:57]: No. We want to add these. I'm surprised they were not here before.

File: src/main/java/edu/wpi/first/gradlerio/deploy/roborio/FRCJNILibraryArtifact.java [hunk 2]
```diff
@@ -74,7 +74,7 @@ public void deploy(DeployContext arg0) {
     }
 
     public FileCollection computeFiles() {
-        Set<File> configFileCaches = configuration.get().getResolvedConfiguration().getFiles();
+        Set<File> configFileCaches = configuration.get().getIncoming().getFiles().getFiles();
```
Comment (Starlight220) [inline src/main/java/edu/wpi/first/gradlerio/deploy/roborio/FRCJNILibraryArtifact.java:77]: `.getFiles().getFiles()`?

File: src/main/java/edu/wpi/first/gradlerio/wpi/dependencies/WPIDependenciesPlugin.java [hunk 3]
```diff
@@ -67,7 +67,7 @@ private Set<File> getFiles() {
         for (Configuration cfg : configs) {
             if (cfg.isCanBeResolved()) {
                 logger.info("Resolving Deps Configuration: " + cfg.getName());
-                files.addAll(cfg.getResolvedConfiguration().getFiles());
+                files.addAll(cfg.getIncoming().getFiles().getFiles());
```
Comment (Starlight220) [inline src/main/java/edu/wpi/first/gradlerio/wpi/dependencies/WPIDependenciesPlugin.java:70]: Ditto

File: src/main/java/edu/wpi/first/gradlerio/wpi/dependencies/tools/WPICppTool.java [hunk 4]
```diff
@@ -18,7 +17,7 @@ public class WPICppTool implements Named {
     private final Provider<String> version;
 
     public WPICppTool(Project project, String name, Provider<String> version, String artifactId, Provider<Directory> toolsFolder) {
-        Configuration config = project.getConfigurations().getByName("wpiCppTools");
+        //Configuration config = project.getConfigurations().getByName("wpiCppTools");
```
Comment (Starlight220) [inline src/main/java/edu/wpi/first/gradlerio/wpi/dependencies/tools/WPICppTool.java:20]: Commented-out code should be removed?

File: src/main/java/edu/wpi/first/gradlerio/deploy/roborio/FRCJNILibraryArtifact.java (hunk 2 above)
Comment (ThadHouse) [reply src/main/java/edu/wpi/first/gradlerio/deploy/roborio/FRCJNILibraryArtifact.java:77]: Yup. Thats the replacement.

File: src/main/java/edu/wpi/first/gradlerio/wpi/dependencies/tools/WPICppTool.java (hunk 4 above)
Comment (ThadHouse) [reply src/main/java/edu/wpi/first/gradlerio/wpi/dependencies/tools/WPICppTool.java:20]: I want to keep this for when we finally getting around to adding installation for C++ tools.

**Review by PeterJohnson — APPROVED** [review]

**Comment by ThadHouse** [comment]
@PeterJohnson I need to merge native uitils and publish that for use here before merging.

**Review by PeterJohnson — APPROVED** [review]

**Pull request merged**